### ISSUE TITLE
post date not considered for lots

### ISSUE THREAD

**Issue by chrpinedo** [ledger/ledger#2011]
Example file:
```
2021/01/01 * Buy stocks
    Assests:Stocks  25 AMZN @ 10$ ; [2021/01/05]
    Assests:Savings
```

The lot date is the transaction date, however it should be post date:
```
$ ledger -f sample.ldg bal stocks --lots
25 AMZN {$10} [2021/01/01]  Assests:Stocks
```

I seems that there some limitation with lot dates. Besides this bug, I also reported that [effective dates are useless with lots](https://github.com/ledger/ledger/issues/1837).

**Comment by tbm** [comment]
A posting date, as the name suggests, is for a *posting*.

The syntax you want is:

```
2021/01/01 * Buy stocks
    Assests:Stocks  25 AMZN [2021/01/05] @ 10$
    Assests:Savings
```

i.e. you specify the *lot* date.

**Comment by tbm** [comment]
> The lot date is the transaction date, however it should be post date:

Oh, I see now what you mean.  I guess you have a point there.

Since the posting has a posting date, the lot date should follow the date of the posting rather than the date of the transaction.

**Comment by chrpinedo** [comment]
Yes, perhaps it could seem to be a strange use, but i was trying to use it in that way

**Comment by petersonev** [comment]
>```
> 2021/01/01 * Buy stocks
>     Assests:Stocks  25 AMZN [2021/01/05] @ 10$
>     Assests:Savings
>```
This syntax also does not seem to work. The lot dates only seem to work when a lot price is explicitly given rather than inferred from the cost amount (or cost total). Seems like that might be a bug but giving the lot price instead of the cost amount seems to work:
```
 2021/01/01 * Buy stocks
     Assests:Stocks  25 AMZN {10$} [2021/01/05]
     Assests:Savings
```

**Comment by tbm** [comment]
> This syntax also does not seem to work.

Works for me:

```
$ cat > d
2021/01/01 * Buy stocks
    Assests:Stocks  25 AMZN [2021/01/05] @ 10$
    Assests:Savings

$ ledger -f d bal --lots
               $-250
25 AMZN {$10} [2021-01-01]  Assests
               $-250    Savings
25 AMZN {$10} [2021-01-01]    Stocks
--------------------
               $-250
25 AMZN {$10} [2021-01-01]
```

**Comment by tbm** [comment]
> This syntax also does not seem to work. The lot dates only seem to work when a lot price is explicitly given rather than inferred from the cost amount (or cost total).

Yes, sorry about that.
